File: Queue/implementation_using_array.py
class Queue:
    def __init__(self, capacity):
        self.front = 0
        self.rear = -1
        self.capacity = capacity
        self.queue = [None] * capacity
    
    def is_full(self) -> bool:
        return (self.rear >= self.capacity - 1)
    
    def is_empty(self) -> bool:
        return (self.rear < self.front)
    
    def enque(self, item):
        if self.is_full():
            print("Queue is full")
            return
        else:
            self.rear = self.rear + 1
            self.queue[self.rear] = item
    
    def deque(self):
        if self.is_empty():
            print("Queue is empty")
            return
        else:
            item = self.queue[self.front]
            self.front = self.front + 1
            return item

        
if __name__ == "__main__":
    queue = Queue(4)
    print(queue.queue)

    queue.enque(5)
    queue.enque(1)
    queue.enque(6)
    queue.enque(4)
    queue.enque(4)

    print(queue.queue)
    
    print(queue.deque())
    print(queue.deque())
    print(queue.deque())
    print(queue.deque())
    print(queue.deque())

File: Queue/implementation_using_array.c
/* Implementing Queue using Array in C */

#include <stdio.h>
#include <stdlib.h>

typedef struct Queue
{
    int front, rear, capacity;
    int *queue;
} Queue;

// Function to initialize the queue
Queue *createQueue(int capacity)
{
    struct Queue *q = (struct Queue *)malloc(sizeof(struct Queue));
    q->capacity = capacity;
    q->front = 0;
    q->rear = -1;
    q->queue = (int *)malloc(q->capacity * sizeof(int));
    return q;
}

void enque(Queue *queue, int item)
{
    if (queue->rear >= queue->capacity)
    {
        printf("Queue is full !");
        return;
    }
    else
    {
        queue->queue[++queue->rear] = item;
    }
}

int deque(Queue *queue)
{
    if (queue->rear < queue->front)
    {
        printf("Queue is empty.");
        return -1;
    }
    else
    {
        return queue->queue[queue->front++];
    }
}

void displayQueue(Queue *queue)
{
    for (int i = queue->front; i <= queue->rear; i++)
    {
        printf("%d\t", queue->queue[i]);
    }
}

int main()
{
    struct Queue *q1 = createQueue(4);
    displayQueue(q1);
    enque(q1, 5);
    enque(q1, 1);
    printf("Deque: %d\n", deque(q1));
    printf("Deque: %d\n", deque(q1));
    printf("Deque: %d\n", deque(q1));
    printf("Deque: %d\n", deque(q1));
    enque(q1, 2);
    displayQueue(q1);

    return 0;
}